src/core_ocean/mode_init/mpas_ocn_init_global_ocean.F

```diff
@@ -998,37 +994,35 @@ subroutine ocn_init_setup_global_ocean_create_model_topo(domain, iErr)!{{{
              call ocn_init_setup_global_ocean_deepen_critical_passages(meshPool, criticalPassagesPool, iErr)
           end if
 
-          ! Fill holes. Enforce that no cell has maxLevelCell that is more than
-          ! 1 vertical level higher than its neighbors.
-          if (config_global_ocean_fill_bathymetry_holes) then
-             call mpas_pool_get_subpool(block_ptr % structs, 'scratch', scratchPool)
-
-             call mpas_pool_get_field(scratchPool, 'smoothedLevels', smoothedLevelsField)
+          ! Alter cells for partial bottom cells before filling in bathymetry
+          ! holes.
+          do iCell = 1, nCells
+             call ocn_alter_bottomDepth_for_pbcs(bottomDepth(iCell), refBottomDepth, maxLevelCell(iCell), iErr)
```

Comment:
@xylar this is now complete. The critical change was the line above. The bathymetry holes would be filled, and then in a later call the cells were altered. Specifically, if the bottom bathymetry is within 5% of the layer thickness to the top of the cell, we round it up and change maxLevelCell to one less. 

My solution here is to call the PBC routine before filling the bathymetry. I left the original call to
`ocn_alter_bottomDepth_for_pbcs`
because it is later in the sequence, and it is within your iterations for SSH for land ice. I thought there was no harm done leaving the later one in place.

Reply:
I agree, I think this is safe to do. I believe I had `ocn_alter_bottomDepth_for_pbcs` in `ocn_init_vertical_grid` because it currently only gets called if you are *not* constraining the vertical grid base don the Haney number, which handles partial bottom cells after making substantial alternations to the grid.  This earlier call might slightly modify the outcome of the later stage but not in a way that I expect to be important (just probably not bit-for-bit).